### PR TITLE
[ci/coverage][4] get coverage data from s3

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
Download the latest coverage data from s3. Attempt to find the right test target to run based on the list of changed files, but I just print the entire coverage data out for now. Finding the right test target is a story for the next PR.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   
```
(base) can@can-Q7M1Q6FVFV ray % python ci/pipeline/ray_release_test.py         
2023-05-05 12:53:55,989:INFO:root:Changed files: ['.buildkite/pipeline.test.yml', 'ci/pipeline/ray_coverage.py', 'ci/pipeline/ray_logger.py', 'ci/pipeline/ray_release_test.py']
2023-05-05 12:53:55,998:INFO:botocore.credentials:Found credentials in environment variables.
2023-05-05 12:53:56,931:INFO:root:Name                                                          Stmts   Miss  Cover
---------------------------------------------------------------------------------
release/__init__.py                                               0      0   100%
release/ray_release/__init__.py                                   0      0   100%
release/ray_release/alerts/__init__.py                            0      0   100%
release/ray_release/alerts/default.py                             7      4    43%
release/ray_release/alerts/handle.py                             21      9    57%
release/ray_release/alerts/long_running_tests.py                 18     18     0%
release/ray_release/alerts/tune_tests.py                         44     44     0%
release/ray_release/alerts/xgboost_tests.py                      34     34     0%
release/ray_release/anyscale_util.py                             30     27    10%
release/ray_release/aws.py                                       45     34    24%
release/ray_release/buildkite/__init__.py                         0      0   100%
release/ray_release/buildkite/concurrency.py                     64     21    67%
release/ray_release/buildkite/filter.py                          45     11    76%
release/ray_release/buildkite/output.py                          13      7    46%
release/ray_release/buildkite/settings.py                       124     44    65%
release/ray_release/buildkite/step.py                            60     21    65%
release/ray_release/cluster_manager/__init__.py                   0      0   100%
release/ray_release/cluster_manager/cluster_manager.py           79     34    57%
release/ray_release/cluster_manager/full.py                      60     20    67%
release/ray_release/cluster_manager/minimal.py                  157     39    75%
release/ray_release/command_runner/__init__.py                    0      0   100%
release/ray_release/command_runner/_anyscale_job_wrapper.py     192     60    69%
release/ray_release/command_runner/_prometheus_metrics.py        76     76     0%
release/ray_release/command_runner/_wait_cluster.py              25     25     0%
release/ray_release/command_runner/anyscale_job_runner.py       150    150     0%
release/ray_release/command_runner/command_runner.py             51     41    20%
release/ray_release/command_runner/job_runner.py                 76     76     0%
release/ray_release/config.py                                   141     64    55%
release/ray_release/env.py                                       20      7    65%
release/ray_release/exception.py                                 85     85     0%
release/ray_release/file_manager/__init__.py                      0      0   100%
release/ray_release/file_manager/file_manager.py                 10      9    10%
release/ray_release/file_manager/job_file_manager.py            104     95     9%
release/ray_release/file_manager/remote_task.py                  42     42     0%
release/ray_release/glue.py                                     258     65    75%
release/ray_release/job_manager/__init__.py                       3      3     0%
release/ray_release/job_manager/anyscale_job_manager.py         182    182     0%
release/ray_release/job_manager/job_manager.py                   70     64     9%
release/ray_release/log_aggregator.py                            57     57     0%
release/ray_release/logger.py                                    11     11     0%
release/ray_release/reporter/__init__.py                          0      0   100%
release/ray_release/reporter/artifacts.py                        31     31     0%
release/ray_release/reporter/db.py                               21     21     0%
release/ray_release/reporter/log.py                              21     21     0%
release/ray_release/reporter/reporter.py                         11      7    36%
release/ray_release/result.py                                    87     62    29%
release/ray_release/scripts/__init__.py                           0      0   100%
release/ray_release/scripts/build_pipeline.py                   110    110     0%
release/ray_release/scripts/get_aws_instance_information.py      25     25     0%
release/ray_release/scripts/get_test_summary.py                  23     23     0%
release/ray_release/scripts/ray_bisect.py                        95     95     0%
release/ray_release/scripts/run_release_test.py                  61     61     0%
release/ray_release/signal_handling.py                           33     23    30%
release/ray_release/template.py                                  82      3    96%
release/ray_release/tests/__init__.py                             0      0   100%
release/ray_release/tests/_test_catch_args.py                     9      0   100%
release/ray_release/tests/_test_run_release_test_sh.py           27      0   100%
release/ray_release/tests/test_alerts.py                         18      1    94%
release/ray_release/tests/test_anyscale_job_manager.py           15     15     0%
release/ray_release/tests/test_anyscale_job_wrapper.py           55      1    98%
release/ray_release/tests/test_bisect.py                         30     30     0%
release/ray_release/tests/test_buildkite.py                     293      4    99%
release/ray_release/tests/test_cluster_manager.py               323     16    95%
release/ray_release/tests/test_config.py                         75      1    99%
release/ray_release/tests/test_env.py                            25      2    92%
release/ray_release/tests/test_glue.py                          367      7    98%
release/ray_release/tests/test_log_aggregator.py                 13     13     0%
release/ray_release/tests/test_result.py                         12     12     0%
release/ray_release/tests/test_run_script.py                     53      2    96%
release/ray_release/tests/test_template.py                       33     33     0%
release/ray_release/tests/test_wheels.py                        132      1    99%
release/ray_release/tests/utils.py                               35     16    54%
release/ray_release/util.py                                      99     71    28%
release/ray_release/wheels.py                                   205     93    55%
release/setup.py                                                  2      2     0%
---------------------------------------------------------------------------------
TOTAL                                                          4775   2281    52%
```

### CODE REVIEW

File: ci/pipeline/ray_release_test.py [hunk 1]
```diff
@@ -13,9 +20,45 @@ def main() -> int:
...
     logger.info(f"Changed files: {changed_files}")
+    test_targets = _get_test_targets_for_changed_files(changed_files)
+    logger.info(test_targets)
     return 0
 
 
+def _get_test_targets_for_changed_files(changed_files: List[str]) -> str:
+    """
+    Get the test target for the changed files.
+    """
```
Comment (aslonnie) [inline ci/pipeline/ray_release_test.py:31]: so.. this is determining what tests to run from a past coverage data? when will this be used? will this also be used on branch/master postmerge or only on pr's?

how about tests that are newly added in a PR? will they get run or not?

Comment (can-anyscale) [reply ci/pipeline/ray_release_test.py:31]: To bootstrap this new approach, I'm thinking starting with at the pr level. branch/master still runs full coverage as they are right now. As we gain trust in this new approach we can optimize further, but fixing pr level itself I think will already save a lot of costs.

There is only one case where this approach will fall short, which you exactly pointed out: newly added test. Newly added test will always get run using existing rules and not affected by coverage.

**Review by aslonnie — APPROVED** [review]

**Comment by can-anyscale** [comment]
save

**Comment by can-anyscale** [comment]
hi

**Comment by can-anyscale** [comment]
three

**Pull request closed without merging**